hibernate-core/src/main/java/org/hibernate/type/descriptor/java/IntegerTypeDescriptor.java

```diff
@@ -110,4 +111,58 @@ public int getDefaultSqlPrecision(Dialect dialect) {
 	public int getDefaultSqlScale() {
 		return 0;
 	}
+
+	@Override
+	public Integer coerce(Object value, CoercionContext coercionContext) {
+		if ( value == null ) {
+			return null;
+		}
+
+		if ( value instanceof Integer ) {
+			return (int) value;
+		}
+
+		if ( value instanceof Short ) {
+			return CoercionHelper.toInteger( (short) value );
```

Comment:
same here and the following method, probably better not cast to primitives

Reply:
Like above, already checked for null